functionary/core/utils/minio.py

```diff
@@ -5,23 +5,40 @@
 from django.http import HttpRequest
 from minio import Minio
 from minio.error import S3Error
+from urllib3.exceptions import MaxRetryError
 from urllib3.response import HTTPResponse
 
 from core.models import Task
 
 
 class MinioInterface:
     def __init__(self, bucket_name: str):
-        self.client = Minio(
-            endpoint=f"{settings.S3_HOST}:{settings.S3_PORT}",
-            access_key=settings.S3_ACCESS_KEY,
-            secret_key=settings.S3_SECRET_KEY,
-            secure=settings.S3_SECURE,
-            region=settings.S3_REGION,
-        )
+        try:
+            self.client = Minio(
+                endpoint=f"{settings.S3_HOST}:{settings.S3_PORT}",
+                access_key=settings.S3_ACCESS_KEY,
+                secret_key=settings.S3_SECRET_KEY,
+                secure=settings.S3_SECURE,
+                region=settings.S3_REGION,
+            )
 
-        self.bucket_name = bucket_name
-        self._create_bucket()
+            self.bucket_name = bucket_name
+            self._create_bucket()
+        except MaxRetryError as err:
```

Comment:
I would catch these here and re-raise a custom exception.   Basically, if I'm instantiating this somewhere (like in the view code below) catching a urlib3 exception or a Minio exception based on the particular error seems like a weird implementation detail.  Instead, let's just define a custom exception at the top of this file and catch and raise.  Also, we should log the error:

```python
except MaxRetryError:
    msg = f"Connection to S3 provider at {settings.S3_HOST}:{settings.S3_PORT} could not be established"
    logger.error(msg)
    raise S3ConnectionError(msg)
except S3Error:
    # similar deal raising same S3ConnectionError
```

This also makes me realize that there's no docstring to this class.  We should add one with the usual class docstring info, plus info about the exception that may be raised.

Reply:
Added the docstring and custom exception